Fix Dictionary.Text not accepting AppSettings.Dictionary

pmme opened the pull request:
Dictionary.DropDownStyle became DropDownList in commit 310a55c3c2 so Dictionary.Items must contain what we want to set in Dictionary.Text.

Fixes #4443.

Changes proposed in this pull request:
 - Preload Dictionary.Items with the current AppSettings.Dictionary if that current dic file exists so that Dictionary.Text will accept the current dictionary.
 - Retain the current dictionary selection during Dictionary_DropDown where previously Dictionary.Items.Clear did make Dictionary.Text invalid and therefore automatically clear the current choice.

I intentionally chose to retain the existing behaviour of loading of the list at the time the drop-down button is clicked rather than loading it as a fixed list at the time the form is created. This enables a user to copy in a new dic file then choose that file from the list while the Settings > Appearance form remains open.

This all comes about because Dictionary.DropDownStyle became DropDownList in commit 310a55c3c2.
 
What did I do to test the code and ensure quality:
 - checked that current dictionary setting is shown when displaying the settings.
 - checked that "None" was shown when the current dic file was missing.
 - checked that current selection is retained when opening the drop down of the combo box.

Has been tested on (remove any that don't apply):
 - Git 2.16.2.windows.1
 - Windows 10
 - GitExtensions on master branch at 5047252f5e

**NOTE:** This pull request (for master) replaces #4626, because that previous pull request was for release/2.51.

gerhardol commented:
This seem to fix the dialog, but is the setting doing anything?
I still have to change manually in Commit

pmme commented:
Interesting... It seems the commit message dictionary choice is stored but is independent of the main settings. The setting that I fixed is modifying the GitExtensions.settings file. In fact, as far as I recall, it always was changing the setting. The problem I fixed was that it had ceased to display it in the combo box.

Currently I have no idea where the dictionary setting for the commit message dialog is stored. When GE is closed and restarted it has remembered the setting but it makes no change in GitExtensions.settings.

I shall investigate further, but not just now...

RussKie commented:
Thank you
